lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -336,15 +338,30 @@ ParseResult FIRParser::parseToken(FIRToken::Kind expectedToken,
   return emitError(message);
 }
 
-/// Parse a list of elements, terminated with an arbitrary token.
+/// Parse a comma-separated list of elements, terminated with an arbitrary
+/// token.
 ParseResult
 FIRParser::parseListUntil(FIRToken::Kind rightToken,
                           const std::function<ParseResult()> &parseElement) {
+  bool first = false;
 
   while (!consumeIf(rightToken)) {
+    // on the second iteration and thereafter, parse a comma.
+    if (first) {
+      if (parseToken(FIRToken::comma, "expected ','")) {
+        return failure();
```

Comment:
This is allowing dangling commas.

Reply:
I backed out of the trailing commas change. Resolved.